test/unit/parallel/mpi_concurrency/mpi_collective_sum_test.cpp

```diff
@@ -270,8 +290,10 @@ TEST_F(MPICollectiveSumTest, ComputeCovarianceComplex) {
     for (int j = i + 1; j < 2 * f.size(); j++)
       covariance_expected(j, i) = covariance_expected(i, j);
 
-  for (int i = 0; i < covariance_test.size(); i++)
-    EXPECT_DOUBLE_EQ(covariance_expected(i), covariance_test(i));
+  for (int i = 0; i < covariance.size(); i++) {
```

Comment:
Maybe explain why the results are the same.

Reply:
I don't think an explanation for why a method called compute A, and one called compute A and B, provide the same A.

Reply:
So do we still need to maintain computeCovariance? Do we ever call with a different estimate than the average?

Reply:
No, not really at the moment. It was written by Andrei in case we need to compute (x-y)(x-y)^t with a generic y != <x>. But currently, there is no test that needs it.